WheelOfFortune/Game.cs

```diff
@@ -91,7 +96,60 @@ public void StartRound()
         /// </summary>
         public void StartTurn()
         {
-            Console.WriteLine(CurrentPuzzle.PuzzleSoFar + "\n");
+            // set new CurrentPlayer if CurrentPlayer is null
+            if(CurrentPlayer == null)
+            {
+                // dequeue from Players list and set to CurrentPlayer aka take next player from queue and set it to current
+                CurrentPlayer = Players.Dequeue();
+            } 
+
+            // so now that we have a current player, they will start to perform guesses so call on StartTurn method from Prompt class passing in CurrentPlayer and CurrentPuzzle
+            Prompt.StartTurn(CurrentPlayer, CurrentPuzzle);
+
+            // the GetAction method from Prompt class should return 1 or 2 from the Enum 
+            char solveOrGuessActionType = Prompt.GetActionType();
+
+            // Take return value from Prompt.GetActionType method and pass in action type to CurrentPlayer.SetAction
+            // This will not return anything, it just sets the action in CurrentAction 
+            CurrentPlayer.SetAction(solveOrGuessActionType);
+
+            // CurrentPlayer.CurrentAction argument is set to an action class either guess or solve puzzle so pass action instance into Prompt action function
+            // I believe it is a string return value so leaving it as string instead of dynamic for now
+            string playerInput = Prompt.PromptAction(CurrentPlayer.CurrentAction);
+
+            // Prompt action will return a valid guess from user (letter or phrase based on what the user selects) so set it to execute function --> playerInput just stores the prompt action call return and pass to execute function
+            CurrentPlayer.CurrentAction.Execute(playerInput, CurrentPuzzle);
+
+            // update Round.Winner to determine a winner
+            // check the current player's current action 
+            // if it is solve puzzle type (from enum return value) execute function returns a boolean (to indicate if they correctly guess or not)
+            // if it returns true then you know you have a round winner
+            // HALP not sure if this is right plz feel free to provide feedback
+            if(CurrentPlayer.CurrentAction.ActionType == SolvePuzzleAction)
+            {
+                bool isPuzzleSolved = CurrentPlayer.PerformAction((Action.ActionType)
+                if(typeof isPuzzleSolved == true)
+                {
+                    Round.Winner = CurrentPlayer;
+                }
+                
+            }
```

Comment:
Here you want to set your `isPuzzleSolved` variable to the return of `CurrentPlayer.CurrentAction.Execute` on line 121 (although it may be better to name it something like `isGuessCorrect` or maybe even `continuePlayerTurn`). If the value is true, then you want to check if the action that the player was taking was the GuessLetterAction or the SolvePuzzleAction, which you could do either by using the ActionTypeProperty that Safy recently added (`CurrentPlayer.CurrentAction.ActionTypeProperty == Action.ActionType.SolvePuzzleAction`) OR by checking the type of the action (`CurrentPlayer.CurrentAction.GetType() == typeof(SolvePuzzleAction` <- this may need to be tested, as I've never actually written this snippet of code before). I think I might have confused you when I mentioned `typeof` earlier, but you can just check that the `isPuzzleSolved` variable is true or false (or just use the variable itself since it's a boolean).

Reply:
Oh good naming convention call out! I was totally confusing myself with the old implementation and forgot that we do not have the CurrentPlayer.Perform action method anymore. Hahaha yes, I was also confused by the typeof comment earlier since I was passively listening in on the teams convo as I was coding this out. I'll also clean up the comments prior to merging ~